test/types/community/collection/filterQuery.test-d.ts

```diff
@@ -236,3 +236,27 @@ expectNotType<Filter<PetModel>>({ type: { $size: 2 } });
 // dot key case that shows it is assignable even when the referenced key is the wrong type
 expectAssignable<Filter<PetModel>>({ 'bestFriend.name': 23 }); // using dot notation permits any type for the key
 expectNotType<Filter<PetModel>>({ bestFriend: { name: 23 } });
+
+// ObjectId are not allowed to be used as a query predicate (issue described here: NODE-3758)
+declare const nonObjectIDCollection: Collection<{ _id: number, otherField: string }>
```

Comment:
What if I have an unparameterized collection? Or a `Collection<Document>`? Do those three keys work or have a type error?

Reply:
If the collection is unparameterized the schema defaults to `Document`.  

In any scenario when `InferIdType` returns `ObjectId` for the `_id` type, there won't be a type error.  The error is only raised in scenarios where the user specifies a particular id type.